.pre-commit-config.yaml

```diff
@@ -1,25 +1,48 @@
+default_stages: [pre-commit, pre-push]
+default_install_hook_types: [pre-commit, pre-push]
 repos:
 - repo: https://github.com/gitleaks/gitleaks
   rev: v8.21.1
   hooks:
   - id: gitleaks
+    stages: [pre-push]
 - repo: https://github.com/koalaman/shellcheck-precommit
   rev: v0.10.0
   hooks:
   - id: shellcheck
     args: ["--external-sources"]
+    stages: [pre-commit]
 - repo: https://github.com/pre-commit/pre-commit-hooks
   rev: v5.0.0
   hooks:
   - id: end-of-file-fixer
+    stages: [pre-commit]
   - id: trailing-whitespace
+    stages: [pre-commit]
   - id: detect-private-key
+    stages: [pre-commit]
 - repo: https://github.com/pylint-dev/pylint
   rev: v3.3.1
   hooks:
   - id: pylint
     exclude: ^hack/boilerplate/boilerplate.py$
+    stages: [pre-commit]
 - repo: https://github.com/jumanjihouse/pre-commit-hooks
   rev: 3.0.0
   hooks:
-    - id: forbid-binary
+  - id: forbid-binary
+    stages: [pre-commit]
+- repo: local
+  hooks:
+  - id: make-verify
+    name: Run make verify
+    entry: make verify
+    language: system
+    stages: [pre-push]
```

Comment:
Do you think we can add a reference to Makefile's `verify` target in here or put up a comment?
A disclaimer saying that make verify is going to invoke a bunch of targets like
`verify-boilerplate verify-modules verify-gen verify-shellcheck verify-conversions verify-tiltfile verify-codespell`

Reply:
Yeah we could use https://pre-commit.com/#hooks-description

Reply:
Great! Let's add some text in here, as a disclaimer, make verify will run `verify-boilerplate verify-modules verify-gen verify-shellcheck verify-conversions verify-tiltfile verify-codespell` all these targets.

Reply:
Sorry for the churn. 
I see that `boilerplate` check is already run as a separate hook up above. 
Do you think it would be beneficial to run these `verify` targets individually as separate hooks and drop redundant hooks like boilerplate and spellcheck ?

Reply:
I see merit in either approach. Though I would lean to running the targets individually because then you can see them running one by one in a terminal rather than just one block you're waiting to finish on. Which option would you prefer?

Reply:
Running them individually as separate hooks at pre-push stage sounds great to me.

Reply:
PR is updated. I think it looks better with each one - 
```
% git push -f
Run make verify-modules..................................................Passed
Run make verify-gen......................................................Passed
Run make verify-shellcheck...............................................Passed
Run make verify-conversions..............................................Passed
Run make verify-tiltfile.................................................Passed
Run make test............................................................Passed
Enumerating objects: 5, done.
...
```

FWIW, the `verify-gen` took a similar amount of time as `test` so maybe `test` doesn't need broken up? 🤷🏻‍♂️

Reply:
Thank you for pointing this out. 
Both `verify-gen` and `test` seem to be running `generate` as a pre-req. 

So maybe we don't break `test` target for backwards compatibility in CI and replace `make test` in pre-commit config with `make go-test-race`. 
We can follow this up in another PR, non blocking comment.